perimeterx/px_context.py

```diff
@@ -501,10 +512,21 @@ def is_monitor_request(self, is_monitor_request):
         self._is_monitor_request = is_monitor_request
 
 
-
 def generate_context_headers(request_headers, sensitive_headers):
     headers = CaseInsensitiveDict()
     for header_name, header_value in request_headers:
         if header_name.lower() not in sensitive_headers:
             headers[header_name] = header_value
     return headers
+
+
+def is_filtered_request(config, request):
+    if sum(1 for _ in filter(lambda whitelist_route_item: re.match(whitelist_route_item, request.path), config.whitelist_routes_regex)) > 0 or \
+           sum(1 for _ in filter(lambda whitelist_route_item: request.path == whitelist_route_item, config.filter_by_route)) > 0:
+        return True
+    if config.filter_by_custom_function:
```

Comment:
Should the custom function be the first condition?

Reply:
No - it should be the last condition in case other (simpler) configurations filter it out first and we can prevent needing to call the function unnecessarily